src/main/java/org/opensearch/ad/task/ADTaskManager.java

```diff
@@ -1538,6 +1544,57 @@ public void updateLatestRealtimeTask(
         }, null, false, listener);
     }
 
+    /**
+     * Update latest realtime task's state, init progress, estimated left minutes and error field.
+     * @param detectorId detector id
+     * @param newState new task state
+     * @param rcfTotalUpdates total updates of RCF model
+     * @param detectorIntervalInMinutes detector interval in minutes
+     * @param error error message
+     */
+    public void updateLatestRealtimeTask(
+        String detectorId,
+        String newState,
+        Long rcfTotalUpdates,
+        Long detectorIntervalInMinutes,
+        String error
+    ) {
+        Float initProgress = null;
+        String state = null;
+        if (detectorIntervalInMinutes != null && rcfTotalUpdates != null) {
+            state = ADTaskState.INIT.name();
+            if (rcfTotalUpdates <= NUM_MIN_SAMPLES) {
+                initProgress = (float) rcfTotalUpdates / NUM_MIN_SAMPLES;
+            } else {
+                state = ADTaskState.RUNNING.name();
+                initProgress = 1.0f;
+            }
+        }
+        if (newState != null) {
+            state = newState;
```

Comment:
What if rcfTotalUpdates < NUM_MIN_SAMPLES and newState is RUNNING?

Reply:
The new state has higher priority. It will always override state calculated with RCF total updates. Will add more comments.